mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVaultHook.java

```diff
@@ -127,7 +127,8 @@ public final int hashCode() {
         Preconditions.checkNotNull(deleteOperation);
 
         return groupMetadataByOwnerAndMessageId(deleteOperation)
-            .flatMap(this::appendToTheVault)
+            .parallel()
+            .concatMap(this::appendToTheVault)
```

Comment:
... I'm dumb but can you explain?

Would this make possible to re-enable some disabled tests?

Reply:
> ... I'm dumb but can you explain?

I am not sure, I think it is a thread ordering issue since in `ParallelFlux` it is a fix parallelism contrary to `Flux.flatMap` which have an ad-hoc parallelism.

> Would this make possible to re-enable some disabled tests?

absolutely